lib/bitcoin/BitcoinRawDataParser.ts

```diff
@@ -0,0 +1,85 @@
+import { Block } from 'bitcore-lib';
+
+/**
+ * Parser for raw bitcoin block data
+ */
+export default class BitcoinRawDataParser {
+
+  /**
+   * The beginning of each block contains the magic bytes indicating main or test net
+   * followed by a 4 byte number indicating how big the block data is
+   */
+  private static magicBytes = {
+    testnet: '0b110907',
+    mainnet: 'f9beb4d9'
+  };
+  private static sizeBytesLength = 4;
+
+  /**
+   * Parse the given raw block data file. It can throw error if block data is invalid when validating magic bytes,
+   * creating new Block, or validating size
+   * @param rawBlockDataFileBuffer The file, in buffer form, to be parsed as blocks
+   */
+  public static parseRawDataFile (rawBlockDataFileBuffer: Buffer): any {
+    let rawBlockDataFileString = rawBlockDataFileBuffer.toString('hex');
+    let hexStrings: string[];
+    let magicBytes: string;
+    const blockMapper: any = {};
+
+    // split the hex by magic bytes
+    if (rawBlockDataFileString.startsWith(BitcoinRawDataParser.magicBytes.testnet)) {
+      magicBytes = BitcoinRawDataParser.magicBytes.testnet;
+      hexStrings = rawBlockDataFileString.split(magicBytes);
+    } else if (rawBlockDataFileString.startsWith(BitcoinRawDataParser.magicBytes.mainnet)) {
+      magicBytes = BitcoinRawDataParser.magicBytes.mainnet;
+      hexStrings = rawBlockDataFileString.split(magicBytes);
+    } else {
+      throw new Error('Invalid block data');
+    }
+
+    // remove the first empty string after split
+    hexStrings.shift();
+
+    let currentHex = '';
+    let count = 0;
+    for (const hexString of hexStrings) {
```

Comment:
My head hurts thinking initially `hexString` can be magic bytes, size, and block data, but realizing it can't be magic bytes.

Is it possible to sorta have each increment of the loop to operates at the [currentHex, magic bytes, block] triple increment? e.g.

foreach (block starting point) {
   magic bytes = ...
   size = ...
   block = ...
   verifySize()...
   new Block()?
}